content/linux-in-cloudnative.md

```diff
@@ -0,0 +1,73 @@
+
+
+I am not claiming to be a Linux expert but have put together a list of places where I've run into issues or seen others struggle. This is by no means exhaustive, and likely doesn't even scratch the surface of where you could potentially run into "legacy" issues while running in the cloud! 
+
+
+
+# Resource Allocations and Limits
+
+Linux has a lot of resource constraints built into the OS and these can cause failures in odd and unexpected ways.
+
+As we saw in the AWS example, process limits exist!
+
+There are also file descriptor limits, inotify limits, port allocation limits, and similar. And for more high performance data centers, we should also keep in mind physical constraints as well (such as disk speed, network latency, heat dissipation, etc...)
+
+For a full list of tuning parameters on SUSE Enterprise Linux and OpenSUSE, check out https://documentation.suse.com/sles/15-SP2/single-html/SLES-tuning/#book-sle-tuning
+
+*TODO* Find more human readable list of parameters?
+
+# Build Process
+
+The processes for building applications don't really change much. Most CI/CD pipelines are really just directed graphs of fancy scripts. This means that the build tooling shouldn't change much other than the way it gets triggered.
+
+We still need to be concerned with aspects like artifact size, build time, governance, and reproducability to be able to best help the Ops team do their job! For example, knowing the dependencies that go into a build and being able to track them to the final product lets us know where potential issues might pop up before they do.
+
+
+# Security! 
+
+NOTE: I'm also not a security expert so the main point is that nothing has changed, and there's just more work to do as Kubernetes is insecure by default. 
+
+Just because the application is walled off in a container doesn't mean there's no threat. There have been many SVEs found where a rouge process can break out of it's confinement like Tai Lung from Kung Fu Panda. This means that we still need to think about security on the host and in the applications as well as security of the cluster data plane and network. 
+
+We need to still be concerned with host and physical security if running locally. Setting up App Armor profiles, correct firewall rules, and minimal privileges are still needed to keep the control plane and applications secure. This also means not running privileged pods in your cluster as those can get root access to your node and mess with kernel parameter (ask me how I know that it's possible...). 
+
+There's also security concerns with the build pipeline. Too many tools take a shortcut and ask for the container to mount the docker socket. This would allows a ci/cd script to make changes to your cluster. One thing that can help this (in my opinion) is moving from the docker daemon to cri-o and using tools that are runtime agnostic.
```

Comment:
```suggestion
There's also security concerns with the build pipeline. Too many tools take a shortcut and ask for the container to mount the docker socket. This would allow a ci/cd script to make changes to your cluster. One thing that can help this (in my opinion) is moving from the docker daemon to cri-o and using tools that are runtime agnostic.
```